src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_params.py

```diff
@@ -0,0 +1,120 @@
+        c.argument('namespace_name', options_list=['--namespace-name'], help='name of the Namespace')
+
+    for scope in ['eventhubs namespace authorization-rule', 'eventhubs namespace authorization-rule keys list', 'eventhubs namespace authorization-rule keys renew']:
+        with self.argument_context(scope) as c:
+            c.argument('authorization_rule_name', arg_type=name_type, id_part='child_name_1', help='name of the Namespace AuthorizationRule')
+            c.argument('namespace_name', options_list=['--namespace-name'], id_part='name', help='name of the Namespace')
+
+    for scope in ['eventhubs namespace authorization-rule create', 'eventhubs namespace authorization-rule update', 'eventhubs eventhub authorization-rule create', 'eventhubs eventhub authorization-rule update']:
+        with self.argument_context(scope) as c:
+            c.argument('rights', arg_type=rights_arg_type)
+
+    with self.argument_context('eventhubs namespace authorization-rule keys renew') as c:
+        c.argument('key_type', arg_type=key_arg_type)
+
+
+# region - Eventhub Create
+    with self.argument_context('eventhubs eventhub') as c:
+        c.argument('event_hub_name', arg_type=name_type, id_part='child_name_1', completer=get_eventhubs_command_completion_list, help='Name of Eventhub')
+
+    for scope in ['eventhubs eventhub create', 'eventhubs eventhub update']:
+        with self.argument_context(scope) as c:
+            c.argument('message_retention_in_days', options_list=['--message-retention'], type=int, help='Number of days to retain events for this Event Hub, value should be 1 to 7 days')
+            c.argument('partition_count', type=int, help='Number of partitions created for the Event Hub, allowed values are from 1 to 32 partitions.')
+            c.argument('status', arg_type=get_enum_type(['Active', 'Disabled', 'SendDisabled', 'ReceiveDisabled']), help='Status of Eventhub')
+            c.argument('enabled', options_list=['--enable-capture'], arg_type=get_three_state_flag(), help='A boolean value that indicates whether capture description is enabled.')
+            c.argument('capture_interval_seconds', arg_group='Capture', options_list=['--capture-interval'], type=int, help='Allows you to set the frequency with which the capture to Azure Blobs will happen, value should between 60 to 900 seconds')
+            c.argument('capture_size_limit_bytes', arg_group='Capture', options_list=['--capture-size-limit'], type=int, help='Defines the amount of data built up in your Event Hub before an capture operation, value should be between 10485760 to 524288000 bytes')
+            c.argument('destination_name', arg_group='Capture-Destination', help='Name for capture destination')
+            c.argument('storage_account_resource_id', arg_group='Capture-Destination', options_list=['--storage-account'], help='Resource id of the storage account to be used to create the blobs')
```

Comment:
This must accept a name or ID of a storage account.

Reply:
the service accepts the storage resource ID only.

Reply:
This is the case for nearly every service. It doesn't change the requirements that the CLI command accept the name or ID.

Reply:
the storage account can be in different resource group

Reply:
implemented, will accept name or ARM ID, help text updated